sdk/v2/core/jobs.go

```diff
@@ -118,9 +118,12 @@ type JobSpec struct {
 	// sidecar container), then logic within those containers must account for
 	// these constraints.
 	SidecarContainers map[string]JobContainerSpec `json:"sidecarContainers,omitempty"` // nolint: lll
-	// TimeoutSeconds specifies the time, in seconds, that must elapse before a
-	// running Job should be considered to have timed out.
-	TimeoutSeconds int64 `json:"timeoutSeconds,omitempty"`
+	// TimeoutDuration specifies the time duration that must elapse before a
+	// running Job should be considered to have timed out. This duration string
+	// is a possibly signed sequence of decimal numbers, each with optional
+	// fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m".
+	// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
```

Comment:
What does it mean to have a negative duration timeout?

Reply:
Good question -- I can't think of a real-world case for this in tandem with how we use this field.  I'll remove that example and ensure the schema only allows non-negative numbers.